lib/src/table/table.service.ts

```diff
@@ -0,0 +1,436 @@
+            }
+
+            if (order) {
+                columnsList = columnsList.sort((a: string, b: string) => {
+                    const firstItem: number = this.tableConfig.order[a] ? this.tableConfig.order[a] : 1;
+                    const secondItem: number = this.tableConfig.order[b] ? this.tableConfig.order[b] : -1;
+                    return firstItem - secondItem;
+                });
+            }
+        }
+
+        this.columnsList = [...columnsList];
+    }
+
+    /**
+     * SETUP TABLE:
+     * Sets up the sorted, paginated and current page table
+     * @param {T[]} table the table
+     * @param {SortInfo} sortInfo The information on how to sort the table: column name, type and asc/desc
+     * @param {number} maxItems The maximum number of items to be displayed per page
+     * @param {number[]} selectedRowIndexes the indexes of the rows selected in the master table
+     */
+    private setupTable(table: T[], sortInfo: SortInfo, maxItems: number, selectedRowIndexes?: number[]): void {
+        const tableWithMeta: TableRowWithMeta<T>[] = table.map((row: T, rowIndex: number) => {
+            return { row, selected: !!selectedRowIndexes?.includes(rowIndex) };
+        });
+        const sortedTableWithMeta: TableRowWithMeta<T>[] = this.makeSortedTable(tableWithMeta, sortInfo);
+        this._sortedTable = [...sortedTableWithMeta.map(item => item.row)];
+
+        const paginatedTableWithMeta: TableRowWithMeta<T>[][] = this.makePaginatedTable(sortedTableWithMeta, maxItems);
+        this._paginatedTable = [...paginatedTableWithMeta.map(item => item.map(x => x.row))];
+
+        this._selectedRows = [...paginatedTableWithMeta.map(item => item.map((x, i) => (x.selected ? i : null)).filter(y => y !== null))];
+
+        if (this._currentPageIndex > this._paginatedTable.length - 1) {
+            this._currentPageIndex = this._paginatedTable.length - 1;
+        }
+        this._currentTable = [...this._paginatedTable[this._currentPageIndex]];
+    }
+
+    /**
+     * MAKE SORTED TABLE
+     * @param table the table you wish to sort with meta data
+     * @param sort the sort info
+     * @returns The new sorted table
+     */
+    private makeSortedTable(table: TableRowWithMeta<T>[], sort?: SortInfo): TableRowWithMeta<T>[] {
+        if (sort) {
+            return this.sortTable(table, sort);
+        }
+        return [...table];
+    }
+
+    /**
+     * MAKE PAGINATED TABLE
+     * @param table the table you wish to paginate with meta data
+     * @param maxItems the number of max items on one page
+     * @returns the paginated table as array or arrays
+     */
+    private makePaginatedTable(table: TableRowWithMeta<T>[], maxItems: number): TableRowWithMeta<T>[][] {
+        const paginatedTable: TableRowWithMeta<T>[][] = [];
+
+        while (table.length > 0) {
+            paginatedTable.push(table.splice(0, maxItems));
+        }
```

Comment:
I think you are using while here like an `if` statement, I think you should just convert it o an if statement
```suggestion
        if (table.length > 0) {
            paginatedTable.push(table.splice(0, maxItems));
        }
```

Reply:
it's mutating the value so the while needs to be there